axiom/nr_php_packages.c

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright 2022 New Relic Corporation. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <math.h>
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "nr_axiom.h"
+#include "nr_limits.h"
+#include "nr_php_packages.h"
+#include "util_memory.h"
+#include "util_minmax_heap.h"
+#include "util_vector.h"
+#include "util_hashmap.h"
+#include "util_hashmap_private.h"
+#include "util_strings.h"
+
+nr_php_package_t* nr_php_package_create(char* name, char* version) {
+  nr_php_package_t* p = NULL;
+
+  if (NULL == name || NULL == version) {
```

Comment:
Is there a spec regarding this?  If we have the package name but can't determine the version, should not report anything or should we still report the package?

Reply:
Good point, the spec says that we should still pass an empty string with a space. I have added this case and added unit tests to ensure that when version is null, we will set the version to an empty string with a space. [2bb60f5](https://github.com/newrelic/newrelic-php-agent/pull/754/commits/2bb60f58ff030adf798b02e29cde14e3ce51e7f0)